scripts/scrapper.ts

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios'
+import * as cheerio from 'cheerio'
+import Meal, { MealCategory } from '#models/meal'
+
+export async function scrapeMenu() {
+  const url = 'https://sks.pwr.edu.pl/menu/'
+
+  const response = await axios.get(url)
+  const $ = cheerio.load(response.data)
+
+  const menuItems: Partial<Meal>[] = []
```

Comment:
i'm not sure if there's need to be that specific in this case but i'm not really familiar with those types and can't tell the difference (would love to hear some explanation)

Reply:
(if you take suggestion below, you can skip it fully)

Reply:
partial means it can be any subset of properties and with pick or omit you take some type and define which if this properties should we pick or omit to create this new type

Reply:
(gpt explanations incoming)

Reply:
In TypeScript, `Partial`, `Pick`, and `Omit` are all utility types, but they serve slightly different purposes when creating new object types by transforming existing ones:

### 1. **Partial**
- `Partial<Type>` makes all properties of `Type` optional.
- This means that any subset of properties is valid, but you don’t have control over which specific properties are included or omitted.
- Example:
  ```typescript
  interface User {
      id: number;
      name: string;
      email: string;
  }

  // Partial<User> allows any subset of properties
  const partialUser: Partial<User> = { name: "Alice" };
  ```

### 2. **Pick**
- `Pick<Type, Keys>` creates a new type by selecting specific properties (keys) from the given type.
- You get to specify exactly which properties you want to retain in the new type, which can be helpful if you need a subset of properties but all are required within that subset.
- Example:
  ```typescript
  interface User {
      id: number;
      name: string;
      email: string;
  }

  // Pick<User, "id" | "name"> retains only `id` and `name` from User
  const pickedUser: Pick<User, "id" | "name"> = { id: 1, name: "Alice" };
  ```

### 3. **Omit**
- `Omit<Type, Keys>` creates a new type by excluding specific properties from the given type.
- This is the inverse of `Pick`, as you specify which properties to remove, keeping all others as they are in the original type.
- Example:
  ```typescript
  interface User {
      id: number;
      name: string;
      email: string;
  }

  // Omit<User, "email"> removes `email`, leaving only `id` and `name`
  const omittedUser: Omit<User, "email"> = { id: 1, name: "Alice" };
  ```

### Summary of Differences
- **`Partial`** allows **any subset of properties**, all optional.
- **`Pick`** allows you to create a type with **only the properties you specify** (all required).
- **`Omit`** lets you create a type by **excluding the properties you specify** (all remaining are required).

Using `Pick` or `Omit` can be more specific than `Partial` when you need only certain properties and want them to stay required.

Reply:
but if you use `.map` the types will be infered so no need to annotate

Reply:
good to know it anyways